crates/bevy_remote/src/lib.rs

```diff
@@ -673,41 +605,41 @@ pub struct BrpMessage {
     pub sender: Sender<BrpResult>,
 }
 
+/// A resource holding the matching sender for the [`BrpReceiver`]'s receiver.
+#[derive(Debug, Resource, Deref, DerefMut)]
+pub struct BrpSender(Sender<BrpMessage>);
+
 /// A resource that receives messages sent by Bevy Remote Protocol clients.
 ///
 /// Every frame, the `process_remote_requests` system drains this mailbox and
 /// processes the messages within.
 #[derive(Debug, Resource, Deref, DerefMut)]
-pub struct BrpMailbox(Receiver<BrpMessage>);
+pub struct BrpReceiver(Receiver<BrpMessage>);
 
-/// A system that starts up the Bevy Remote Protocol server.
-fn start_server(mut commands: Commands, address: Res<HostAddress>, remote_port: Res<HostPort>) {
+fn setup_mailbox_channel(mut commands: Commands) {
     // Create the channel and the mailbox.
-    let (request_sender, request_receiver) = channel::bounded(CHANNEL_SIZE);
-    commands.insert_resource(BrpMailbox(request_receiver));
-
-    IoTaskPool::get()
-        .spawn(server_main(address.0, remote_port.0, request_sender))
-        .detach();
+    let (request_sender, request_receiver) = async_channel::bounded(CHANNEL_SIZE);
+    commands.insert_resource(BrpSender(request_sender));
+    commands.insert_resource(BrpReceiver(request_receiver));
 }
 
-/// A system that receives requests placed in the [`BrpMailbox`] and processes
+/// A system that receives requests placed in the [`BrpReceiver`] and processes
 /// them, using the [`RemoteMethods`] resource to map each request to its handler.
 ///
 /// This needs exclusive access to the [`World`] because clients can manipulate
 /// anything in the ECS.
 fn process_remote_requests(world: &mut World) {
-    if !world.contains_resource::<BrpMailbox>() {
+    if !world.contains_resource::<BrpReceiver>() {
         return;
     }
 
-    while let Ok(message) = world.resource_mut::<BrpMailbox>().try_recv() {
+    while let Ok(message) = world.resource_mut::<BrpReceiver>().try_recv() {
         // Fetch the handler for the method. If there's no such handler
         // registered, return an error.
         let methods = world.resource::<RemoteMethods>();
 
         let Some(handler) = methods.0.get(&message.method) else {
-            let _ = message.sender.send_blocking(Err(BrpError {
+            let _ = message.sender.force_send(Err(BrpError {
```

Comment:
Can you explain the rationale for the change to `force_send`?

Reply:
I think this ensures errors are reported even if the queue is full. This will overwrite regular messages to ensure the error message is always sent. Did I get that right @LiamGallagher737 ?

Reply:
The previous `send_blocking` is not available on WASM. Another option is to use [`try_send`](https://docs.rs/async-channel/latest/async_channel/struct.Sender.html#method.try_send) but I don't think it is necessary as the channel is written to from in this function and only once. I could add some docs to support this.